src/plugins/data/common/exports/export_csv.tsx

```diff
@@ -61,14 +82,36 @@ export function datatableToCSV(
   }
 
   // Convert the array of row objects to an array of row arrays
-  const csvRows = rows.map((row) => {
-    return sortedColumnIds.map((id) =>
-      escapeValues(raw ? row[id] : formatters[id].convert(row[id]))
-    );
-  });
+  const csvRows = rows
+    .map((row) => {
+      return sortedColumnIds.map((id) =>
+        escapeValues(raw ? row[id] : formatters[id].convert(row[id]))
+      );
+    })
+    .sort(rowSortPredicate(sortedColumnIds, columnSorting));
 
   return (
     [header, ...csvRows].map((row) => row.join(csvSeparator)).join(LINE_FEED_CHARACTER) +
     LINE_FEED_CHARACTER
   ); // Add \r\n after last line
 }
+
+function rowSortPredicate(
+  sortedColumnIds: string[],
+  columnSorting?: EuiDataGridColumnSortingConfig[]
+) {
+  if (!columnSorting) return () => 0;
+
+  const columnIdMap = new Map(columnSorting.map(({ id }) => [id, sortedColumnIds.indexOf(id)]));
+  return (rowA: string[], rowB: string[]) => {
```

Comment:
Have you considered leveraging the `getSortingCriteria` from the `@kbn/sort-predicates` package used by the Lens datatable? This will ensure consistency between the two.

Can find an example usage here: `x-pack/plugins/lens/public/visualizations/datatable/components/table_actions.ts`

Reply:
This was a great call, took a bit of time to wrangle all the necessary inputs but finally got it working. Updated in 44baf770e2d1e31ee54df6c5fd07448839f5b8ac.